py-arrow: add v1.3.0 (switch to flit-core)

wdconinc opened the pull request:
This PR adds py-arrow v1.3.0, which switches from setuptools to flit-core as build system.

wdconinc committed:
py-arrow: add v1.3.0 (switch to flit-core)

adamjstewart commented:
Does this actually work for you? I have build errors: https://github.com/arrow-py/arrow/issues/1187

wdconinc commented:
```
13:56:01 wdconinc@menelaos ~/git/DD4hep ((tags/v01-30) $%) $ spack env activate --temp 
spack install --add py-arrow
==> Created and activated temporary environment in /tmp/spack-zf3liqbl
13:57:12 wdconinc@menelaos ~/git/DD4hep ((tags/v01-30) $%) $ spack install --add py-arrow
==> Concretized 1 spec
[+]  k24vsey  py-arrow@1.3.0%gcc@14.2.0 build_system=python_pip arch=linux-ubuntu24.10-skylake
[+]  exckd52      ^gcc-runtime@14.2.0%gcc@14.2.0 build_system=generic arch=linux-ubuntu24.10-skylake
[e]  kauwo5s      ^glibc@2.40%gcc@14.2.0 build_system=autotools arch=linux-ubuntu24.10-skylake
[+]  g4tyzss      ^py-flit-core@3.9.0%gcc@14.2.0 build_system=python_pip arch=linux-ubuntu24.10-skylake
[+]  p3xwhhc      ^py-pip@24.1.1%gcc@14.2.0 build_system=generic arch=linux-ubuntu24.10-skylake
[+]  rbj34k2      ^py-python-dateutil@2.9.0.post0%gcc@14.2.0 build_system=python_pip arch=linux-ubuntu24.10-skylake
[+]  peoakzs          ^py-setuptools@70.2.0%gcc@14.2.0 build_system=generic arch=linux-ubuntu24.10-skylake
[+]  4rfekvd          ^py-setuptools-scm@8.1.0%gcc@14.2.0+toml build_system=python_pip arch=linux-ubuntu24.10-skylake
[+]  m5ovwds              ^git@2.45.2%gcc@14.2.0+man+nls+perl+subtree~svn~tcltk build_system=autotools arch=linux-ubuntu24.10-skylake
[+]  y7gs5ho                  ^autoconf@2.72%gcc@14.2.0 build_system=autotools arch=linux-ubuntu24.10-skylake
[+]  3ilsqjg                  ^automake@1.16.5%gcc@14.2.0 build_system=autotools arch=linux-ubuntu24.10-skylake
[+]  56i26hz                  ^curl@8.10.1%gcc@14.2.0~gssapi~ldap~libidn2~librtmp~libssh~libssh2+nghttp2 build_system=autotools libs=shared,static tls=openssl arch=linux-ubuntu24.10-skylake
[+]  7ql3jds                      ^nghttp2@1.63.0%gcc@14.2.0 build_system=autotools arch=linux-ubuntu24.10-skylake
[+]  2thqyzc                  ^libidn2@2.3.7%gcc@14.2.0 build_system=autotools arch=linux-ubuntu24.10-skylake
[+]  zigmw2r                      ^libunistring@1.2%gcc@14.2.0 build_system=autotools arch=linux-ubuntu24.10-skylake
[+]  h4tkyti                  ^libtool@2.4.7%gcc@14.2.0 build_system=autotools arch=linux-ubuntu24.10-skylake
[+]  bmr63nc                      ^findutils@4.10.0%gcc@14.2.0 build_system=autotools patches=440b954 arch=linux-ubuntu24.10-skylake
[+]  ntvvhg2                  ^m4@1.4.19%gcc@14.2.0+sigsegv build_system=autotools patches=9dc5fbd,bfdffa7 arch=linux-ubuntu24.10-skylake
[+]  nqbpzxl                      ^libsigsegv@2.14%gcc@14.2.0 build_system=autotools arch=linux-ubuntu24.10-skylake
[+]  lqtwp3w                  ^openssh@9.8p1%gcc@14.2.0+gssapi build_system=autotools arch=linux-ubuntu24.10-skylake
[+]  k7sbr4c                      ^krb5@1.21.3%gcc@14.2.0+shared build_system=autotools arch=linux-ubuntu24.10-skylake
[+]  ipm6adk                          ^bison@3.8.2%gcc@14.2.0~color build_system=autotools arch=linux-ubuntu24.10-skylake
[+]  n6fvy6b                      ^libedit@3.1-20240808%gcc@14.2.0 build_system=autotools arch=linux-ubuntu24.10-skylake
[+]  oby3z7a                  ^pcre2@10.44%gcc@14.2.0~jit+multibyte build_system=autotools arch=linux-ubuntu24.10-skylake
[+]  52xuuzz              ^py-packaging@24.1%gcc@14.2.0 build_system=python_pip arch=linux-ubuntu24.10-skylake
[+]  ncrxjxr              ^py-tomli@2.0.1%gcc@14.2.0 build_system=python_pip arch=linux-ubuntu24.10-skylake
[+]  v4x2inq              ^py-typing-extensions@4.12.2%gcc@14.2.0 build_system=python_pip arch=linux-ubuntu24.10-skylake
[+]  wjietzb          ^py-six@1.16.0%gcc@14.2.0 build_system=python_pip arch=linux-ubuntu24.10-skylake
[+]  xh3jm25      ^py-types-python-dateutil@2.8.19.14%gcc@14.2.0 build_system=python_pip arch=linux-ubuntu24.10-skylake
[+]  k5rqc7a      ^py-wheel@0.43.0%gcc@14.2.0 build_system=generic arch=linux-ubuntu24.10-skylake
[+]  luhfpk7      ^python@3.11.9%gcc@14.2.0+bz2+crypt+ctypes+dbm~debug+libxml2+lzma~nis~optimizations+pic+pyexpat+pythoncmd+readline+shared+sqlite3+ssl~tkinter+uuid+zlib build_system=generic patches=13fa8bf,b0615b2,ebdca64,f2fd060 arch=linux-ubuntu24.10-skylake
[+]  yvkx54j          ^bzip2@1.0.8%gcc@14.2.0~debug~pic+shared build_system=generic arch=linux-ubuntu24.10-skylake
[+]  tmfraoc              ^diffutils@3.10%gcc@14.2.0 build_system=autotools arch=linux-ubuntu24.10-skylake
[+]  olqn3vz          ^expat@2.6.3%gcc@14.2.0+libbsd build_system=autotools arch=linux-ubuntu24.10-skylake
[+]  7fzdygc              ^libbsd@0.12.2%gcc@14.2.0 build_system=autotools arch=linux-ubuntu24.10-skylake
[+]  cghdlkr                  ^libmd@1.1.0%gcc@14.2.0 build_system=autotools arch=linux-ubuntu24.10-skylake
[+]  wd2wyop          ^gdbm@1.23%gcc@14.2.0 build_system=autotools arch=linux-ubuntu24.10-skylake
[+]  flgrqsz          ^gettext@0.22.5%gcc@14.2.0+bzip2+curses+git~libunistring+libxml2+pic+shared+tar+xz build_system=autotools arch=linux-ubuntu24.10-skylake
[+]  z6fudon              ^libiconv@1.17%gcc@14.2.0 build_system=autotools libs=shared,static arch=linux-ubuntu24.10-skylake
[+]  7ns5e25              ^libxml2@2.10.3%gcc@14.2.0+pic~python+shared build_system=autotools arch=linux-ubuntu24.10-skylake
[+]  yg55wxn              ^tar@1.34%gcc@14.2.0 build_system=autotools zip=pigz arch=linux-ubuntu24.10-skylake
[+]  y3vrmq5                  ^pigz@2.8%gcc@14.2.0 build_system=makefile arch=linux-ubuntu24.10-skylake
[+]  qylfxak                  ^zstd@1.5.6%gcc@14.2.0+programs build_system=makefile compression=none libs=shared,static arch=linux-ubuntu24.10-skylake
[+]  zqxcss2          ^gmake@4.4.1%gcc@14.2.0~guile build_system=generic arch=linux-ubuntu24.10-skylake
[+]  hlqhlvc          ^libffi@3.4.6%gcc@14.2.0 build_system=autotools arch=linux-ubuntu24.10-skylake
[+]  znld5fh          ^libxcrypt@4.4.36%gcc@14.2.0~obsolete_api build_system=autotools arch=linux-ubuntu24.10-skylake
[+]  whrl74h              ^perl@5.40.0%gcc@14.2.0+cpanm+opcode+open+shared+threads build_system=generic arch=linux-ubuntu24.10-skylake
[+]  zjua53y                  ^berkeley-db@18.1.40%gcc@14.2.0+cxx~docs+stl build_system=autotools patches=26090f4,b231fcc arch=linux-ubuntu24.10-skylake
[+]  wxnybky          ^ncurses@6.5%gcc@14.2.0~symlinks+termlib abi=none build_system=autotools patches=7a351bc arch=linux-ubuntu24.10-skylake
[+]  sekbs2r          ^openssl@3.3.1%gcc@14.2.0~docs+shared build_system=generic certs=mozilla arch=linux-ubuntu24.10-skylake
[-]  n2q2656              ^ca-certificates-mozilla@2023-05-30%gcc@14.2.0 build_system=generic arch=linux-ubuntu24.10-skylake
[+]  xugnc5t          ^pkgconf@2.2.0%gcc@14.2.0 build_system=autotools arch=linux-ubuntu24.10-skylake
[+]  2cn2put          ^readline@8.2%gcc@14.2.0 build_system=autotools patches=bbf97f1 arch=linux-ubuntu24.10-skylake
[+]  adzcdp3          ^sqlite@3.46.0%gcc@14.2.0+column_metadata+dynamic_extensions+fts~functions+rtree build_system=autotools arch=linux-ubuntu24.10-skylake
[+]  xuaqrwj          ^util-linux-uuid@2.40.2%gcc@14.2.0 build_system=autotools arch=linux-ubuntu24.10-skylake
[+]  q7bgryy          ^xz@5.4.6%gcc@14.2.0~pic build_system=autotools libs=shared,static arch=linux-ubuntu24.10-skylake
[+]  v7o32a5          ^zlib-ng@2.2.1%gcc@14.2.0+compat+new_strategies+opt+pic+shared build_system=autotools arch=linux-ubuntu24.10-skylake
[+]  de6wmv5      ^python-venv@1.0%gcc@14.2.0 build_system=generic arch=linux-ubuntu24.10-skylake

[+] /usr (external glibc-2.40-kauwo5srhv2gu563iw7ftgfy4ve4qxuj)
[+] /opt/software/linux-ubuntu24.10-skylake/gcc-14.2.0/gcc-runtime-14.2.0-exckd52o7l27fe6ocxl2jxes5tmdtelo
[+] /opt/software/linux-ubuntu24.10-skylake/gcc-14.2.0/util-linux-uuid-2.40.2-xuaqrwjhtypb2d653byrbvbr2wf6sh2v
[+] /opt/software/linux-ubuntu24.10-skylake/gcc-14.2.0/xz-5.4.6-q7bgryyy2jffm5mm4dxazegs7q3bbamr
[+] /opt/software/linux-ubuntu24.10-skylake/gcc-14.2.0/libiconv-1.17-z6fudonvawf55k5fp6k3roujcyovmydt
[+] /opt/software/linux-ubuntu24.10-skylake/gcc-14.2.0/ncurses-6.5-wxnybkylctkt3u2xwlj2swdzuhvx4ijl
[+] /opt/software/linux-ubuntu24.10-skylake/gcc-14.2.0/zlib-ng-2.2.1-v7o32a5acvaewnuiun6gxvnndluoxfzy
[+] /opt/software/linux-ubuntu24.10-skylake/gcc-14.2.0/zstd-1.5.6-qylfxakuaxlfcrjh2lwf3kmv7ym3md3u
[+] /opt/software/linux-ubuntu24.10-skylake/gcc-14.2.0/libffi-3.4.6-hlqhlvck5yxxis3gkk5pnqtqarql75hc
[+] /opt/software/linux-ubuntu24.10-skylake/gcc-14.2.0/libmd-1.1.0-cghdlkrugr5y2clen26w6pc3y5so3tpy
[+] /opt/software/linux-ubuntu24.10-skylake/gcc-14.2.0/bzip2-1.0.8-yvkx54jus2drwj7rp4zh3mt3jfmzs3z7
[+] /opt/software/linux-ubuntu24.10-skylake/gcc-14.2.0/libxcrypt-4.4.36-znld5fhsqnzgu3vtgxerd4zgbdcrml3y
[+] /opt/software/linux-ubuntu24.10-skylake/gcc-14.2.0/readline-8.2-2cn2putytyg5syicggpdzmhvtwj4p4g6
[+] /opt/software/linux-ubuntu24.10-skylake/gcc-14.2.0/openssl-3.3.1-sekbs2rafopilqfhw23b5k3vajlmhnz3
[+] /opt/software/linux-ubuntu24.10-skylake/gcc-14.2.0/pigz-2.8-y3vrmq5kfqxwrrcfi3kdaerrc2v43eh3
[+] /opt/software/linux-ubuntu24.10-skylake/gcc-14.2.0/libxml2-2.10.3-7ns5e25gajitjj5k3r7xtzb326lynzr4
[+] /opt/software/linux-ubuntu24.10-skylake/gcc-14.2.0/libbsd-0.12.2-7fzdygcc25xtxnpei4bbmjaqxjhpna7u
[+] /opt/software/linux-ubuntu24.10-skylake/gcc-14.2.0/gdbm-1.23-wd2wyop6qijlzrc4jjqw3vni3huo2ar3
[+] /opt/software/linux-ubuntu24.10-skylake/gcc-14.2.0/sqlite-3.46.0-adzcdp3ecygr3uuhwoofmqlq3mznso7t
[+] /opt/software/linux-ubuntu24.10-skylake/gcc-14.2.0/tar-1.34-yg55wxnaqmqrssdoyuimnnrxzo3zqngo
[+] /opt/software/linux-ubuntu24.10-skylake/gcc-14.2.0/expat-2.6.3-olqn3vzu3wd6dejrs3gbytzf5o7c5ar4
[+] /opt/software/linux-ubuntu24.10-skylake/gcc-14.2.0/gettext-0.22.5-flgrqsz2mgzlbusqvxbnbfq4kbj63btt
[+] /opt/software/linux-ubuntu24.10-skylake/gcc-14.2.0/python-3.11.9-luhfpk766tfj5t7vdxw3da773imck22h
[+] /opt/software/linux-ubuntu24.10-skylake/gcc-14.2.0/python-venv-1.0-de6wmv5stlmm4jfrscojypt4elrkhsyo
[+] /opt/software/linux-ubuntu24.10-skylake/gcc-14.2.0/py-types-python-dateutil-2.8.19.14-xh3jm25kjxxq3veo3kvm7j4ptwji2vwj
[+] /opt/software/linux-ubuntu24.10-skylake/gcc-14.2.0/py-six-1.16.0-wjietzburzpyrr4iracmv64k6ccxp4cc
[+] /opt/software/linux-ubuntu24.10-skylake/gcc-14.2.0/py-python-dateutil-2.9.0.post0-rbj34k255dxquojfli6izln37mzmwdhl
[+] /opt/software/linux-ubuntu24.10-skylake/gcc-14.2.0/py-arrow-1.3.0-k24vseyojuio4fafto42ptimk3h55msy
==> Updating view at /tmp/spack-zf3liqbl/.spack-env/view
13:58:23 wdconinc@menelaos ~/git/DD4hep ((tags/v01-30) $%) $ python
Python 3.11.9 (main, Oct 13 2024, 12:01:58) [GCC 14.2.0] on linux
Type "help", "copyright", "credits" or "license" for more information.
>>> import arrow
>>> arrow.__version__
'1.3.0'
>>> 
```

adamjstewart commented:
Interesting. The build fails for me unless I already have arrow installed in my PYTHONPATH.

wdconinc commented:
I was able to build and run inside `docker pull --rm -it spack/ubuntu-noble:develop`, with externals for speedup:
```
root@f7a35a331ca8:~# spack external find python 
==> The following specs have been detected on this system and added to /root/.spack/packages.yaml
python@3.12.3
root@f7a35a331ca8:~# spack external find git
==> The following specs have been detected on this system and added to /root/.spack/packages.yaml
git@2.43.0
root@f7a35a331ca8:~# spack spec py-arrow
Input spec
--------------------------------
 -   py-arrow

Concretized
--------------------------------
 -   py-arrow@1.3.0%gcc@13.2.0 build_system=python_pip arch=linux-ubuntu24.04-skylake
[+]      ^gcc-runtime@13.2.0%gcc@13.2.0 build_system=generic arch=linux-ubuntu24.04-skylake
[e]      ^glibc@2.39%gcc@13.2.0 build_system=autotools arch=linux-ubuntu24.04-skylake
 -       ^py-flit-core@3.9.0%gcc@13.2.0 build_system=python_pip arch=linux-ubuntu24.04-skylake
[+]      ^py-pip@23.1.2%gcc@13.2.0 build_system=generic arch=linux-ubuntu24.04-skylake
 -       ^py-python-dateutil@2.8.2%gcc@13.2.0 build_system=python_pip arch=linux-ubuntu24.04-skylake
 -           ^py-setuptools@69.2.0%gcc@13.2.0 build_system=generic arch=linux-ubuntu24.04-skylake
 -           ^py-setuptools-scm@8.0.4%gcc@13.2.0+toml build_system=python_pip arch=linux-ubuntu24.04-skylake
[e]              ^git@2.43.0%gcc@13.2.0+man+nls+perl+subtree~svn~tcltk build_system=autotools arch=linux-ubuntu24.04-skylake
 -               ^py-packaging@24.1%gcc@13.2.0 build_system=python_pip arch=linux-ubuntu24.04-skylake
 -               ^py-tomli@2.0.1%gcc@13.2.0 build_system=python_pip arch=linux-ubuntu24.04-skylake
 -               ^py-typing-extensions@4.12.2%gcc@13.2.0 build_system=python_pip arch=linux-ubuntu24.04-skylake
 -           ^py-six@1.16.0%gcc@13.2.0 build_system=python_pip arch=linux-ubuntu24.04-skylake
 -       ^py-types-python-dateutil@2.8.19.14%gcc@13.2.0 build_system=python_pip arch=linux-ubuntu24.04-skylake
 -       ^py-wheel@0.41.2%gcc@13.2.0 build_system=generic arch=linux-ubuntu24.04-skylake
[e]      ^python@3.12.3%gcc@13.2.0+bz2+crypt+ctypes+dbm~debug+libxml2+lzma~nis~optimizations+pic+pyexpat~pythoncmd+readline+shared+sqlite3+ssl~tkinter+uuid+zlib build_system=generic arch=linux-ubuntu24.04-skylake
[+]      ^python-venv@1.0%gcc@13.2.0 build_system=generic arch=linux-ubuntu24.04-skylake

root@f7a35a331ca8:~# spack install py-arrow
...
==> Installing py-arrow-1.3.0-i4ezzxwypbbls6e7w5f25fb37hcvqzai [17/17]
==> No binary for py-arrow-1.3.0-i4ezzxwypbbls6e7w5f25fb37hcvqzai found: installing from source
==> Fetching https://mirror.spack.io/_source-cache/archive/d4/d4540617648cb5f895730f1ad8c82a65f2dad0166f57b75f3ca54759c4d67a85.tar.gz
==> No patches needed for py-arrow
==> py-arrow: Executing phase: 'install'
==> py-arrow: Successfully installed py-arrow-1.3.0-i4ezzxwypbbls6e7w5f25fb37hcvqzai
  Stage: 0.46s.  Install: 2.41s.  Post-install: 0.07s.  Total: 3.15s
[+] /opt/spack/opt/spack/linux-ubuntu24.04-skylake/gcc-13.2.0/py-arrow-1.3.0-i4ezzxwypbbls6e7w5f25fb37hcvqzai
root@f7a35a331ca8:~# spack load py-arrow
root@f7a35a331ca8:~# python
Python 3.12.3 (main, Sep 11 2024, 14:17:37) [GCC 13.2.0] on linux
Type "help", "copyright", "credits" or "license" for more information.
>>> import arrow
>>> print(arrow.__version__)
1.3.0
```
